src/de/grobox/liberario/fragments/DirectionsFragment.java

```diff
@@ -240,6 +244,23 @@ public void onResume() {
 		displayFavouriteTrips();
 		refreshAutocomplete(false);
 
+		// check if there's a preference for selected products for the current network
+		TransportNetwork current_network = Preferences.getTransportNetwork(getContext());
+		String network_pref = current_network.getId().toString() + "_selected_products";
+		SharedPreferences sp = getContext().getSharedPreferences(Preferences.NETWORK_SETTINGS, 0);
+		// get the selected products from the pref
+		Set<String> string_set = sp.getStringSet(network_pref, new HashSet<String>());
+        // if there is a pref for the current network, read the selected products and update
+        if(string_set.size()>0) {
```

Comment:
I think it might be better to use `null` as the default parameter and check for `!= null` here. Then no `HashSet` needs to be initialized.